### PR TITLE
fix: better language detection in migration/setup workflow

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1547 (epic, all others are sub tasks)
- Closes [UNIFY-1452](https://cypress-io.atlassian.net/browse/UNIFY-1452)
- Closes [UNIFY-1581](https://cypress-io.atlassian.net/browse/UNIFY-1581)
- Closes [UNIFY-1444](https://cypress-io.atlassian.net/browse/UNIFY-1444)
- Closes [UNIFY-1666](https://cypress-io.atlassian.net/browse/UNIFY-1666)

### User facing changelog

 
- Improve language detection throughout app (JS or TS).
- Remove ability to select JS/TS when configuring a project.

### Additional details

Allowing the user to select JS/TS during project setup proved to add a lot of complexity and additional surface area for bugs. For this reason, we decided to **remove this feature** and **automatically** detect JS/TS, defaulting to JS if we are unsure. 

The [full technical brief is here](https://github.com/cypress-io/tech-briefs/pull/22) and explains the how and why behind this decision. You should read it before reviewing this PR.

The detected language (JS or TS) is used for initial scaffolding (`cypress.config.js`, etc), migrating (`cypress.json` -> `cypress.config`) and the type of file we create when scaffolding a new spec from within the runner.

### How has the user experience changed?

- No longer show a "select language" screen
- Identify language type based on algorithm (see [technical brief](https://github.com/cypress-io/tech-briefs/pull/22))

This screen is gone:

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/launchpad/cypress/e2e/project-setup.cy.ts [hunk 1]
```diff
@@ -347,92 +333,7 @@ describe('Launchpad: Setup Project', () => {
         verifyScaffoldedFiles('e2e')
       })
 
-      it('can setup e2e testing for a project selecting TS when CT is configured and config file is JS', () => {
```
Comment (lmiller1990) [inline packages/launchpad/cypress/e2e/project-setup.cy.ts:350]: You cannot do this anymore, you are either using JS or TS - this feature never made any sense in the first place.

File: packages/launchpad/cypress/e2e/project-setup.cy.ts [hunk 2]
```diff
@@ -506,23 +403,6 @@ describe('Launchpad: Setup Project', () => {
         cy.contains('Project Setup')
       })
 
-      it('can reconfigure config after e2e has been set up', () => {
-        scaffoldAndOpenProject('pristine-with-e2e-testing')
-        cy.withCtx((ctx) => {
-          ctx.coreData.forceReconfigureProject = {
```
Comment (lmiller1990) [inline packages/launchpad/cypress/e2e/project-setup.cy.ts:512]: We don't support "reconfigure" anymore... I checked the code base, we can remove all code related to `forceReconfigureProject`. I didn't want to bundle that refactor here, though.

**Comment by jennifer-shehane** [comment]
@lmiller1990 Looks like there's conflicts in this branch now.

**Review by estrada9166 — APPROVED** [review]

File: packages/scaffold-config/src/detect.ts [hunk 3]
```diff
@@ -64,3 +71,83 @@ export function detect (projectPath: string): DetectFramework {
...
+      return 'ts'
+    }
+
+    if (fs.existsSync(path.join(projectRoot, 'cypress.config.js'))) {
+      debug('Detected cypress.config.js - using JS')
+
+      return 'js'
+    }
+  } catch (e) {
+    //
```
Comment (ryanthemanuel) [inline packages/scaffold-config/src/detect.ts:120]: Worth debug logging this error?

**Review by ryanthemanuel — APPROVED** [review]

File: packages/scaffold-config/src/detect.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/scaffold-config/src/detect.ts]: Sure, https://github.com/cypress-io/cypress/pull/21220/commits/a8cdfb10f2dea22a41165f15e8dc453def876d44

Also removed superfluous try/catch.

**Pull request merged**